src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/Model/Field.cs

```diff
@@ -5,7 +5,27 @@
 
 namespace CSharpSyntaxGenerator
 {
-    public class Field
+    public class TreeTypeChild
+    {
+    }
+
+    public class Choice : TreeTypeChild
+    {
+        [XmlElement(ElementName = "Field", Type = typeof(Field))]
+        [XmlElement(ElementName = "Choice", Type = typeof(Choice))]
+        [XmlElement(ElementName = "Sequence", Type = typeof(Sequence))]
+        public List<TreeTypeChild> Children;
+    }
+
+    public class Sequence : TreeTypeChild
+    {
+        [XmlElement(ElementName = "Field", Type = typeof(Field))]
+        [XmlElement(ElementName = "Choice", Type = typeof(Choice))]
+        [XmlElement(ElementName = "Sequence", Type = typeof(Sequence))]
```

Comment:
>Sequence [](start = 35, length = 8)

Does it make sense to have a sequence nested in a sequence?

Reply:
Good point!

Reply:
Resolved!